### PR TITLE
Assorted dependency updates

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -60,16 +56,21 @@ flate2 = "1"
 fs_at.workspace = true
 git-testament = "0.2"
 home = "0.5.4"
-lazy_static.workspace = true
 libc = "0.2"
-once_cell = { workspace = true, optional = true }
+once_cell.workspace = true
 opener = "0.6.0"
+# Used by `curl` or `reqwest` backend although it isn't imported by rustup:
```
Comment (rami3l) [inline Cargo.toml]: I see what you mean here, but I'm afraid for someone without any context of this change, these comments might seem to refer to `openssl` and some other stuff below.

The following should be able to limit the scope of this comment to `openssl`:

```suggestion
# `openssl` is used by `curl` or `reqwest` backend although it isn't imported by rustup:
```

File: src/dist/triple.rs [hunk 2]
```diff
@@ -68,15 +68,17 @@ impl PartialTargetTriple {
...
                 LIST_ARCHS.join("|"),
                 LIST_OSES.join("|"),
                 LIST_ENVS.join("|")
-            );
-            static ref RE: Regex = Regex::new(&PATTERN).unwrap();
-        }
+            )
+        });
+
+        static RE: Lazy<Regex> = Lazy::new(|| Regex::new(&PATTERN).unwrap());
```
Comment (rami3l) [inline src/dist/triple.rs]: Maybe we can inline `PATTERN` as long as it's only used for `RE`'s initialization? The current form gives me the impression that the former is used elsewhere...

File: Cargo.toml (hunk 1 above)
Comment (djc) [reply Cargo.toml]: Fair, fixed as suggested!

File: src/dist/triple.rs (hunk 2 above)
Comment (djc) [reply src/dist/triple.rs]: Yeah, good idea -- took care of that for both instances of this pattern.

**Review by rami3l — APPROVED** [review]
Oops, just got sidetracked by another issue for a while. LGTM!

**Pull request merged**